### PR TITLE
Wip/d3locale

### PR DESCRIPTION
**Pull request by MilosKozak**

### CODE REVIEW

File: lib/d3locales.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  time: '%H:%M:%S',
+  periods: ['AM', 'PM'],
+  days: ['Neděle', 'Pondělí', 'Úterý', 'Středa', 'Čtvrtek', 'Pátek', 'Sobota'],
+  shortDays: ['Ne', 'Po', 'Út', 'St', 'Čt', 'Pá', 'So'],
+  months: ['Leden', 'Únor', 'Březen', 'Duben', 'Květen', 'Červen', 'Červenec', 'Srpen', 'Září', 'Říjen', 'Listopad', 'Prosinec'],
+  shortMonths: ['Led', 'Úno', 'Bře', 'Dub', 'Kvě', 'Čvn', 'Čvc', 'Srp', 'Zář', 'Říj', 'Lis', 'Pro']
+};
+
+d3locales.de_DE = {
+  decimal: ",",
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l26)

File: lib/d3locales.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  thousands: ".",
+  grouping: [3],
+  currency: ["", " €"],
+  dateTime: "%A, der %e. %B %Y, %X",
+  date: "%d.%m.%Y",
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l31)

File: lib/d3locales.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+  months: ['Leden', 'Únor', 'Březen', 'Duben', 'Květen', 'Červen', 'Červenec', 'Srpen', 'Září', 'Říjen', 'Listopad', 'Prosinec'],
+  shortMonths: ['Led', 'Úno', 'Bře', 'Dub', 'Kvě', 'Čvn', 'Čvc', 'Srp', 'Zář', 'Říj', 'Lis', 'Pro']
+};
+
+d3locales.de_DE = {
+  decimal: ",",
+  thousands: ".",
+  grouping: [3],
+  currency: ["", " €"],
+  dateTime: "%A, der %e. %B %Y, %X",
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l30)

File: lib/d3locales.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+d3locales.fi_FI = {
+  decimal: ",",
+  thousands: "\xa0",
+  grouping: [3],
+  currency: ["", "\xa0€"],
+  dateTime: "%A, %-d. %Bta %Y klo %X",
+  date: "%-d.%-m.%Y",
+  time: "%H:%M:%S",
+  periods: ["a.m.", "p.m."],
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l78)

File: lib/d3locales.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  thousands: ".",
+  grouping: [3],
+  currency: ["", " €"],
+  dateTime: "%A, der %e. %B %Y, %X",
+  date: "%d.%m.%Y",
+  time: "%H:%M:%S",
+  periods: ["AM", "PM"], // unused
+  days: ["Sonntag", "Montag", "Dienstag", "Mittwoch", "Donnerstag", "Freitag", "Samstag"],
+  shortDays: ["So", "Mo", "Di", "Mi", "Do", "Fr", "Sa"],
+  months: ["Januar", "Februar", "März", "April", "Mai", "Juni", "Juli", "August", "September", "Oktober", "November", "Dezember"],
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l36)

File: lib/d3locales.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+  thousands: "\xa0",
+  grouping: [3],
+  currency: ["", "\xa0€"],
+  dateTime: "%A, %-d. %Bta %Y klo %X",
+  date: "%-d.%-m.%Y",
+  time: "%H:%M:%S",
+  periods: ["a.m.", "p.m."],
+  days: ["sunnuntai", "maanantai", "tiistai", "keskiviikko", "torstai", "perjantai", "lauantai"],
+  shortDays: ["Su", "Ma", "Ti", "Ke", "To", "Pe", "La"],
+  months: ["tammikuu", "helmikuu", "maaliskuu", "huhtikuu", "toukokuu", "kesäkuu", "heinäkuu", "elokuu", "syyskuu", "lokakuu", "marraskuu", "joulukuu"],
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l81)

File: lib/d3locales.js [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+d3locales.fi_FI = {
+  decimal: ",",
+  thousands: "\xa0",
+  grouping: [3],
+  currency: ["", "\xa0€"],
+  dateTime: "%A, %-d. %Bta %Y klo %X",
+  date: "%-d.%-m.%Y",
+  time: "%H:%M:%S",
+  periods: ["a.m.", "p.m."],
+  days: ["sunnuntai", "maanantai", "tiistai", "keskiviikko", "torstai", "perjantai", "lauantai"],
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l79)

File: lib/d3locales.js [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+  periods: ['AM', 'PM'],
+  days: ['Neděle', 'Pondělí', 'Úterý', 'Středa', 'Čtvrtek', 'Pátek', 'Sobota'],
+  shortDays: ['Ne', 'Po', 'Út', 'St', 'Čt', 'Pá', 'So'],
+  months: ['Leden', 'Únor', 'Březen', 'Duben', 'Květen', 'Červen', 'Červenec', 'Srpen', 'Září', 'Říjen', 'Listopad', 'Prosinec'],
+  shortMonths: ['Led', 'Úno', 'Bře', 'Dub', 'Kvě', 'Čvn', 'Čvc', 'Srp', 'Zář', 'Říj', 'Lis', 'Pro']
+};
+
+d3locales.de_DE = {
+  decimal: ",",
+  thousands: ".",
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l27)

File: lib/d3locales.js [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+  months: ['enero', 'febrero', 'marzo', 'abril', 'mayo', 'junio', 'julio', 'agosto', 'septiembre', 'octubre', 'noviembre', 'diciembre'],
+  shortMonths: ['ene', 'feb', 'mar', 'abr', 'may', 'jun', 'jul', 'ago', 'sep', 'oct', 'nov', 'dic']
+};
+
+d3locales.fi_FI = {
+  decimal: ",",
+  thousands: "\xa0",
+  grouping: [3],
+  currency: ["", "\xa0€"],
+  dateTime: "%A, %-d. %Bta %Y klo %X",
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l75)

File: lib/d3locales.js [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+  grouping: [3],
+  currency: ["", " €"],
+  dateTime: "%A, der %e. %B %Y, %X",
+  date: "%d.%m.%Y",
+  time: "%H:%M:%S",
+  periods: ["AM", "PM"], // unused
+  days: ["Sonntag", "Montag", "Dienstag", "Mittwoch", "Donnerstag", "Freitag", "Samstag"],
+  shortDays: ["So", "Mo", "Di", "Mi", "Do", "Fr", "Sa"],
+  months: ["Januar", "Februar", "März", "April", "Mai", "Juni", "Juli", "August", "September", "Oktober", "November", "Dezember"],
+  shortMonths: ["Jan", "Feb", "Mrz", "Apr", "Mai", "Jun", "Jul", "Aug", "Sep", "Okt", "Nov", "Dez"]
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Codacy Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1013683211/issues/source?bid=2452379&fileBranchId=2875367#l37)

File: lib/d3locales.js [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+
+d3locales.fi_FI = {
+  decimal: ',',
+  thousands: '\xa0',
+  grouping: [3],
+  currency: ['', '\xa0€'],
+  dateTime: '%A, %-d. %Bta %Y klo %X',
+  date: '%-d.%-m.%Y',
+  time: '%H:%M:%S',
+  periods: ['a.m.', 'p.m.'],
```
Comment (jasoncalabrese) [inline lib/d3locales.js]: Don't think this is right for `fi`

Comment (MilosKozak) [inline lib/d3locales.js]: Fi is taken from d3 original code. Let's put it in dev and people should add/correct their locales.

**Comment by sulkaharo** [comment]
LGTM -> timeline day and month is correctly localized for me in Finnish.

**Comment by jasoncalabrese** [comment]
When I tried it in Finnish I saw a.m. and p.m.

**Comment by MilosKozak** [comment]
No am pm saw in FI

**Comment by MilosKozak** [comment]
@Sulkaharo ?

**Comment by jasoncalabrese** [comment]
I bet I tested it in 12 hour mode, still seems strange to have the different am/pm format for Fi when in 12 hour format

**Comment by MilosKozak** [comment]
There was am, pm in most localizations with comment unused. Maybe it's not completely true

**Pull request merged**